File: sdk/go/examples/stub.go
// Stub client example — unit-testing with dependency injection.
//
// Demonstrates NewStubPachcaClient with a custom MessagesService implementation.
//
// Usage:
//
//	go run stub.go
package main

import (
	"context"
	"fmt"
	"log"

	pachca "github.com/pachca/openapi/sdk/go/generated"
)

// fakeMessages implements pachca.MessagesService with a canned response.
type fakeMessages struct {
	pachca.MessagesServiceStub
}

func (f *fakeMessages) GetMessage(_ context.Context, _ int32) (*pachca.Message, error) {
	return &pachca.Message{
		ID:      1,
		Content: "fake message",
	}, nil
}

func main() {
	client := pachca.NewStubPachcaClient(
		pachca.WithStubMessages(&fakeMessages{}),
	)

	msg, err := client.Messages.GetMessage(context.Background(), 1)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("Got: %q (id=%d)\n", msg.Content, msg.ID)
}

File: sdk/kotlin/examples/stub.kt
/**
 * Stub client example — unit-testing with dependency injection.
 *
 * Demonstrates PachcaClient.stub() with a custom MessagesService override.
 *
 * Usage:
 *
 *   kotlin stub.kt
 */
package examples.stub

import com.pachca.sdk.PachcaClient
import com.pachca.sdk.MessagesService
import com.pachca.sdk.Message
import com.pachca.sdk.MessageEntityType
import kotlinx.coroutines.runBlocking
import java.time.OffsetDateTime

class FakeMessages : MessagesService {
    override suspend fun getMessage(id: Int): Message {
        return Message(
            id = 1,
            entityType = MessageEntityType.DISCUSSION,
            entityId = 1,
            chatId = 1,
            rootChatId = 1,
            content = "fake message",
            userId = 1,
            createdAt = OffsetDateTime.now(),
            url = "",
            files = emptyList(),
        )
    }
}

fun main() = runBlocking {
    val client = PachcaClient.stub(messages = FakeMessages())

    val msg = client.messages.getMessage(1)
    println("Got: \"${msg.content}\" (id=${msg.id})")

    client.close()
}
